### ISSUE TITLE
script ignores key based login

### ISSUE THREAD

**Issue by adieball** [laurent22/rsync-time-backup#136]
I'm sure I just oversee something obvious.

I have a user on "ns1" called tmbackup.
That users ssh key has been put onto the remote server

using default ssh, I can login to that server without password using the key (which has no passphrase)

Using the same user in the script gives me the following:

```
sudo rsync-tmbackup.sh /etc tmbackup@192.168.119.10:ns1
tmbackup@192.168.119.10's password: 
tmbackup@192.168.119.10's password: 
tmbackup@192.168.119.10's password: 
rsync-tmbackup: Safety check failed - the destination does not appear to be a backup folder or drive (marker file not found).
rsync-tmbackup: If it is indeed a backup folder, you may add the marker file by running the following command:
rsync-tmbackup: 
rsync-tmbackup: ssh -p 22 tmbackup@192.168.119.10 'mkdir -p -- "ns1" ; touch "ns1/backup.marker"'
rsync-tmbackup: 
```

It continuously asks for a password which makes the script unusable using cron to directly copy via SSH

**Comment by Thr3d** [comment]
Enabling debug for ssh will give you an idea whats happening with the key. This can be done by adding `-v` to the SSH_CMD.

Example:
```
> grep -n 'SSH_CMD="ssh' rsync_tmbackup.sh 
149:		SSH_CMD="ssh -v -p $SSH_PORT ${SSH_USER}@${SSH_HOST}"
156:		SSH_CMD="ssh -v -p $SSH_PORT ${SSH_USER}@${SSH_HOST}"

```

**Comment by pink-ink** [comment]
I have to enter the password **11 times**, then the script is finished successfully.
I also thought it's an endless loop of entering the password again and again.
But it seems, that for every command in the script which is sent via ssh, I have to enter the password again.
Is there any way to improve this, so I have to enter the password only once?

**Comment by pink-ink** [comment]
I've found a solution, where I have to enter the ssh password only once.
It's about connection reuse.
"SSHv2 allows the same authenticated connection to establish multiple 'channels' – interactive shell, batch command, SFTP, along with the secondary ones such as agent-forwarding or TCP-forwarding. Your server probably supports connection multiplexing by default. (If your admins complain, it's not caching your password anywhere – it's caching the whole connection.)"

It's described here in detail:
[Cache the password if SSH-keys are forbidden](https://superuser.com/questions/1362894/cache-the-password-if-ssh-keys-are-forbidden)

Basically I've added the last 3 lines to the file ~/.ssh/config, which now looks like this:
Host 192.168.1.5
 HostkeyAlgorithms +ssh-dss
 AddKeysToAgent yes
 UseKeychain yes
 IdentityFile ~/.ssh/id_rsa
 ControlPath ~/.ssh/S.%r@%h:%p
 ControlMaster auto
 ControlPersist 15m

It can also be
Host *
for all hosts.

**Comment by taw00** [comment]
In my .ssh/config file I have:
```
host workhorse
    hostname 192.168.1.5
    user todd
    IdentityFile ~/.ssh/id_rsa
```

I do a . . .
```
rsync-tmbackup.sh $HOME todd@workhorse:backup-folder $HOME/my-excludes-file.txt
```
And everything works as expected.

@laurent22 , I would close this as not a bug. He is either not using this SSH correctly, or has something misconfigured.

**Comment by laurent22** [comment]
Ok let's close then.
